Tools/LyTestTools/ly_test_tools/o3de/multi_test_framework.py

```diff
@@ -361,6 +363,122 @@ class MultiTestCollector(pytest.Class):
         Custom pytest collector which programmatically adds test functions based on data in the MultiTestSuite class
         """
 
+        # Decorator function to add extra lookup information for the test functions
+        @staticmethod
+        def _set_marks(marks: dict) -> typing.Callable:
+            """
+            Sets pytest marks for a MultiTest test
+
+            :param marks: The pytest marks to set as a dict
+            :return: A function to set the marks
+            """
+            def spec_impl(func):
+                @functools.wraps(func)
+                def inner(*args, **argv):
+                    return func(*args, **argv)
+
+                inner.marks = marks
+                return inner
+
+            return spec_impl
+
+        @staticmethod
+        def _make_single_run(inner_test_spec: AbstractTestBase) -> typing.Callable:
+            """
+            Prepares a test run to be collected as a SingleTest. Single tests are ran independently of each other as
+            opposed to a SharedTest. This also includes setup() and teardown() functions if applicable.
+
+            :param inner_test_spec: The type of test (i.e. SingleTest, SharedTest)
```

Comment:
Given the param, I assume this allows non-SingleTest tests to be run as if they were a SingleTest. Might be worth explicitly noting in the docstring, if that's the case.

...if it's not the case, then the param docstring is incorrect! Also, this appears to slightly incorrect, as it expects the actual test object that will be run, not only its type.

Reply:
Fixed docstring to be more explicit